### PR TITLE
Dvmapply

### PR DESCRIPTION
**Pull request by snschau**
# Summary of changes
add some yamil block code for section images.

## Frontend
- add dvm-application-offers.js, dvm-application-process.js, dvm-application-requirements.js, and dvm-application-selection.js to add the images. 

## Backend
- none

# Test Plan

Insert steps. Include URLs of Gatsby Cloud test site and Drupal multidev.

https://sstest.gatsbyjs.io/dvm-program-application - should have 4 full width images/with text overlay

### CODE REVIEW

File: src/components/blocks/ovc/heading-layout.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React from "react"
```
Comment (mmafe) [inline src/components/blocks/ovc/heading-layout.js:1]: Should this file be part of the PR?

File: src/components/blocks/ovc/dvm-application-offers.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import React from "react"
```
Comment (mmafe) [inline src/components/blocks/ovc/dvm-application-offers.js:1]: It looks like all four components could be accomplished with one component. Recommend having one block component called dvm-application.js, then pass the colourOptions and icon (i.e., fa-circle-1) as variables in yamlWidget.js.

**Review by mmafe — CHANGES_REQUESTED** [review]

File: src/components/shared/yamlWidget.js [hunk 3]
```diff
@@ -20,7 +24,11 @@ const YamlWidget = (props) => {
         'international_explore_btns': <InternationalExploreButtons />,
         'international_explore_grid': <InternationalExploreGrid />,
         'international_talk_current_student':<InternationalTalkCurrentStudent />,
-        'international_explore_lead':<InternationalExploreLead />
+        'international_explore_lead':<InternationalExploreLead />,
+        'dvm_application_selection':<DVMApplicationSelection />,
```
Comment (mmafe) [inline src/components/shared/yamlWidget.js:28]:
```suggestion
        'dvm_application_selection':<DVMApplication colourOptions={background: "var(--uog-yellow)", colour: "#000000"} icon="fa-circle-3" />,
```

Example of passing variable through to component

File: src/components/blocks/ovc/dvm-application-selection.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          path {
+            alias
+          }
+        }
+      }
+    }
+  }
+`
+
+export default function DVMApplicationSelection() {
```
Comment (mmafe) [inline src/components/blocks/ovc/dvm-application-selection.js:68]:
```suggestion
export default function DVMApplication (props) {
```
Example if we changed this file to dvm-application.js and passed props to component

File: src/components/blocks/ovc/dvm-application-selection.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  return <StaticQuery query={query} render={({blockContentYamlBlock}) => render(blockContentYamlBlock, colourOptions)} />
```
Comment (mmafe) [inline src/components/blocks/ovc/dvm-application-selection.js:69]:
```suggestion
  return <StaticQuery query={query} render={({blockContentYamlBlock}) => render(blockContentYamlBlock, props.colourOptions, props.icon)} />
```

File: src/components/blocks/ovc/dvm-application-selection.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    return null;
+  }
+  
+  return (
+    <>
+        <div className="d-flex flex-column bg-light">
+          <Overlay.GatsbyImage gatsbyImageData={getImage(yamlFiles[yamlMap.image.src])} alt={yamlMap.image.alt}>
+            <Container className="page-container">
+              <Row className="h-100 w-100 p-5 justify-content-center align-items-center">
+                  <p className="display-2 text-dark"> <i class="fa-light fa-circle-3 fa-lg"> </i> <strong>{yamlMap.title}</strong></p>
```
Comment (mmafe) [inline src/components/blocks/ovc/dvm-application-selection.js:32]:
```suggestion
                  <p className="display-2 text-dark"> <i className=`fa-light {props.icon} fa-lg`> </i> <strong>{yamlMap.title}</strong></p>
```

File: src/components/blocks/ovc/dvm-application-process.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import {getImage} from "gatsby-plugin-image"
+import Overlay from "components/shared/overlay"
+import styled from 'styled-components'
+
+const yaml = require('js-yaml');
+
+const Shadow = styled.p`
+  text-shadow: 0px 0px 4px #ffffff;
+`
+const colourOptions = {background: "var(--uog-red)", colour: "#ffffff"};
```
Comment (mmafe) [inline src/components/blocks/ovc/dvm-application-process.js:13]: Is this variable being used? If not, then we could just pass in the icon.

**Review by mmafe — CHANGES_REQUESTED** [review]

**Comment by snschau** [comment]
will have to go with individual files - as each of these references a different yaml block in drupal - thus, have different images and headings.  The differences are not just style (as in the international pages for talk to a student).

**Pull request closed without merging**